plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3004,13 +3005,19 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             Map<IcebergColumnHandle, Domain> newEnforced = new LinkedHashMap<>();
             Map<IcebergColumnHandle, Domain> newUnenforced = new LinkedHashMap<>();
             Map<IcebergColumnHandle, Domain> domains = predicate.getDomains().orElseThrow(() -> new VerifyException("No domains"));
-            domains.forEach((columnHandle, domain) -> {
+            domains.forEach( (columnHandle, domain) -> {
+                List<Domain> newDomains;
                 if (!isConvertableToIcebergExpression(domain)) {
                     unsupported.put(columnHandle, domain);
                 }
                 else if (canEnforceColumnConstraintInSpecs(typeManager.getTypeOperators(), icebergTable, partitionSpecIds, columnHandle, domain)) {
                     newEnforced.put(columnHandle, domain);
                 }
+                else if (!(newDomains = canEnforcePartialColumnConstraintInSpecs(typeManager.getTypeOperators(), icebergTable, partitionSpecIds, columnHandle, domain)).isEmpty()) {
```

Comment:
Could you confirm Iceberg library if we can simplify this transform logic? https://github.com/apache/iceberg/tree/main/api/src/main/java/org/apache/iceberg/expressions

I mean updating `ExpressionConverter` like this instead of converting the value in `applyFilter`. 
```java
greaterThanOrEqual(Expressions.day("ts"), Literal.of(ts).to(Types.DateType.get()))
```

Just an idea, I haven't verified the feasibility yet.